src/test/java/com/jfb/tickets/service/UserServiceTests.java

```diff
@@ -0,0 +1,178 @@
+package com.jfb.tickets.service;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import com.jfb.tickets.exceptions.EmptyNameFieldException;
+import com.jfb.tickets.exceptions.ResourceNotFoundException;
+import com.jfb.tickets.exceptions.UserUpdateNotAllowedException;
+import com.jfb.tickets.model.Ticket;
+import com.jfb.tickets.model.User;
+import com.jfb.tickets.repository.TicketRepository;
+import com.jfb.tickets.repository.UserRepository;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceTests {
+
+  @Mock private UserRepository userRepository;
+
+  @Mock private TicketRepository ticketRepository;
+
+  @InjectMocks private UserService userService;
+
+  private UUID userId;
+  private User user;
+
+  @BeforeEach
+  public void setup() {
+    userId = UUID.randomUUID();
+    user = new User("Test User", true);
+    ReflectionTestUtils.setField(userService, "allowUserUpdateAndTicketCreate", true);
```

Comment:
try to avoid reflection. instead of that create usual object:
```java
@Mock private UserRepository userRepository;
@Mock private TicketRepository ticketRepository;
private UserService userService;
...

  @BeforeEach
  public void setup() {
    ...
    userService = new UserService(userRepository, userRepository, true);
}
```